Lib/multiprocessing/synchronize.py

```diff
@@ -123,21 +123,89 @@ def _make_name():
         return '%s-%s' % (process.current_process()._config['semprefix'],
                           next(SemLock._rand))
 
+if sys.platform == 'darwin':
+    #
+    # Specific MacOSX Semaphore
+    #
+
+    class _MacOSXSemaphore(SemLock):
```

Comment:
Wouldn't it be better to handle the SemLock class differently at the C level?

Reply:
> Wouldn't it be better to handle the SemLock class differently at the C level?

As I mentioned in the problem, I'm waiting for feedback (from the Mac OS team ?) before going any further. 

A C implementation is an option, even if it seems more complicated than Python. 
That said, I'm available to go deep into the SemLock C code

Reply:
Actually, since the SemLock class is implemented in C, it's preferrable to first patch it there IMO. But let's CC some people I know they are on macOS: @ned-deily @ronaldoussoren @freakboy3742.

Reply:
I can't profess any particular expertise in this area. However, given the underlying implementation is written in C, I agree that it makes some sense for the implementation of this workaround to also be in C (rather than have an underlying implementation that returns NotImplemented, wrapped by a layer that overrides that.

Reply:
> rather than have an underlying implementation that returns NotImplemented, wrapped by a layer that overrides that.

The main idea is to listen to the `acquire` and `release` methods to update a shared counter, and to return the counter value when the `get_value` method is invoked.

I am going to open a new PR with a workaround written in C.